src/theory/arith/congruence_manager.cpp

```diff
@@ -81,14 +81,13 @@ void ArithCongruenceManager::finishInit(eq::EqualityEngine* ee)
   if (options().arith.arithEqSolver)
   {
     // use our own copy
-    d_allocEe.reset(
-        new eq::EqualityEngine(d_notify, context(), "arithCong::ee", true));
+    d_allocEe.reset(new eq::EqualityEngine(
+        d_env, context(), d_notify, "arithCong::ee", true));
```

Comment:
Have you considered using the following?
```suggestion
    d_allocEe = std::make_unique<eq::EqualityEngine>(
        d_env, context(), d_notify, "arithCong::ee", true);
```

I'm aware we have this pattern all over our code base, but imho we should consider modernizing this.